test/hotspot/jtreg/compiler/predicates/assertion/TestAssertionPredicates.java

```diff
@@ -0,0 +1,471 @@
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates NoTieredCompilation
+ */
+
+/*
+ * @test id=Xcomp
+ * @bug 8288981 8350579
+ * @run main/othervm -Xcomp
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=inline,compiler.predicates.TestAssertionPredicates::inline
+ *                   compiler.predicates.TestAssertionPredicates Xcomp
+ */
+
+/*
+ * @test id=LoopMaxUnroll0
+ * @bug 8288981 8350579
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xcomp -XX:LoopMaxUnroll=0
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates LoopMaxUnroll0
+ */
+
+/*
+ * @test id=StressXcomp
+ * @bug 8288981 8350579
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xcomp -XX:+UnlockDiagnosticVMOptions -XX:+StressIGVN -XX:+StressGCM -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates Stress
+ */
+
+/*
+ * @test id=StressXbatch
+ * @bug 8288981 8350579
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xbatch -XX:+UnlockDiagnosticVMOptions -XX:+StressIGVN -XX:+StressGCM -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates Stress
+ */
+
+/*
+ * @test id=NoLoopPredication
+ * @bug 8288981 8350579
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xbatch -XX:-UseLoopPredicate
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates NoLoopPredication
+ */
+
+/*
+ * @test id=NoFlags
+ * @bug 8288981 8350579
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
```

Comment:
Can you explain why you are enabling `AbortVMOnCompilationFailure`?

Reply:
I added an additional comment further up in the test to explain the reason.